src/util-napatech.c

```diff
@@ -277,29 +685,34 @@ static void *NapatechStatsLoop(void *arg)
             break;
         }
 
-        UpdateStreamStats(tv, hInfo, hStatStream,
-                stream_cnt, stream_config, streamCounters);
+        UpdateStreamStats(tv, hInfo, hstat_stream,
+                stream_cnt, stream_config, total_counters,
+                dispatch_host, dispatch_drop, dispatch_fwd,
+                is_inline, enable_stream_stats,
+                stream_counters);
+
+#ifdef NAPATECH_ENABLE_BYPASS
+        if (bypass_supported) {
+            UpdateFlowStats(tv, hInfo, hstat_stream, flow_counters, 0);
+        }
+#endif /* NAPATECH_ENABLE_BYPASS */
 
         StatsSyncCountersIfSignalled(tv);
         usleep(1000000);
     }
 
     /* CLEAN UP NT Resources and Close the info stream */
     if ((status = NT_InfoClose(hInfo)) != NT_SUCCESS) {
-        NT_ExplainError(status, error_buffer, sizeof (error_buffer) - 1);
-        SCLogError(SC_ERR_RUNMODE, "NT_InfoClose() failed: %s\n", error_buffer);
+        NAPATECH_ERROR(SC_ERR_NAPATECH_INIT_FAILED, status);
```

Comment:
When are the buffers allocated for `stream_counters` freed if an error occurs here?

Reply:
I'm confused by this comment:  stream_counters is created on the stack.  There is no need to free it.  Am I missing something?

Reply:
Please disregard my comment -- my error.